packages/frontend/src/context/MyContext.tsx

```diff
@@ -102,6 +119,47 @@ export const MyContextProvider = (
     address: address as `0x${string}`,
   });
 
+  const getStores = async () => {
+    const stores = new Map();
+
+    const logs = await publicClient.getLogs({
+      address: abi.addresses.ShopReg as Address,
+      event: parseAbiItem(
+        "event Transfer(address indexed from, address indexed to, uint256 value)",
+      ),
+    });
+    const _stores = logs[0].topics;
+    await Promise.all(
+      _stores.map(async (id) => {
+        try {
+          const ownerAdd = (await publicClient.readContract({
```

Comment:
The logs already have have owner `to` in them (`address indexed to`) you should be able to extract this infromation from the logs directly without having to query the contract again.